src/com/xilinx/rapidwright/timing/TimingModel.java

```diff
@@ -500,17 +502,82 @@ protected List<TimingGroup> determineGroups(List<Node> nodes, List<IntentCode> n
         return result;
     }
 
+    private Pair<Integer,Integer> findReferenceTileLocation(Device d) {
+        class Helper {
+            Map<Integer, Set<Integer>> spans;
+            public Helper() {
+                spans = new HashMap<Integer,Set<Integer>>();
+            }
+            void insert(int val, int span) {
+                if (!spans.containsKey(span)) {
+                    spans.put(span, new HashSet<Integer>());
+                }
+                spans.get(span).add(val);
+            }
+            Set<Integer> longest() {
+                List<Integer> keys = new ArrayList<>(spans.keySet());
+                Integer maxSpan = Collections.max(keys);
+                return spans.get(maxSpan);
+            }
+        }
+
+
+        // for each column, look for valid row
+        Helper colHelper = new Helper();
+        for (int i = 0; i < d.getColumns(); i++) {
+            int span = 0;
+            for (int j = 0; j < d.getRows(); j++) {
+                if (d.getTile("INT_X" + i + "Y" + j) != null) {
+                    span++;
+                }
+            }
+            colHelper.insert(i,span);

```

Comment:
```suggestion
        for (int x = 0; x < d.getColumns(); x++) {
            int span = 0;
            for (int y = 0; y < d.getRows(); y++) {
                if (d.getTile("INT", x, y) != null) {
                    span++;
                }
            }
            colHelper.insert(x,span);

```
Refactor iterator names to match x,y.  Using more efficient tile getter.